src/reporting.cc

```diff
@@ -159,3 +158,106 @@ void PrintReport(const snowcrash::Report& report,
         PrintAnnotation("warning:", *it, source, isUseLineNumbers);
     }
 }
+
+template <typename T>
+T* GetMemberValue(const refract::IElement::MemberElementCollection& collection, const std::string& key) {
+    refract::IElement::MemberElementCollection::const_iterator i = collection.find(key);
+    if (i == collection.end()) {
+        return NULL;
+    }
+
+    return refract::TypeQueryVisitor::as<T>((*i)->value.second);
+}
+
+
+struct AnnotationToString {
+
+    std::vector<size_t> linesEndIndex;
+    const bool useLineNumbers;
+
+    AnnotationToString(const std::string& source, const bool useLineNumbers) : useLineNumbers(useLineNumbers) {
+        if (useLineNumbers) {
+            GetLinesEndIndex(source, linesEndIndex);
+        }
+    }
+
+    const std::string operator()(const refract::IElement* annotation) {
+        std::stringstream output;
+
+        if (annotation->element() != "annotation") {
+            return output.str();
+        }
+
+        if (const refract::ArrayElement* classes = GetMemberValue<refract::ArrayElement>(annotation->meta, "classes")) {
+            if (classes->value.size() == 1) {
```

Comment:
Agree, but it is question what to do. 
There is more similar places in code while conversion annotation to string